pysr/regressor_sequence.py

```diff
@@ -0,0 +1,189 @@
+    if recursive_history_length is not None and recursive_history_length <= 0:
+        raise ValueError(
+            "The `recursive_history_length` parameter must be greater than 0 (otherwise it's not recursion)."
+        )
+    if len(X.shape) > 2:
+        raise ValueError(
+            "Recursive symbolic regression only supports up to 2D data; please flatten your data first"
+        )
+    elif len(X) < 2:
+        raise ValueError(
+            "Recursive symbolic regression requires at least 2 datapoints; if you tried to pass a 1D array, use array.reshape(-1, 1)"
+        )
+    if len(X) <= recursive_history_length + 1:
+        raise ValueError(
+            f"Recursive symbolic regression with a history length of {recursive_history_length} requires at least {recursive_history_length + 2} datapoints."
+        )
+    if isinstance(weights, np.ndarray) and len(weights) != len(X):
+        raise ValueError("The length of `weights` must have shape (n_times,).")
+    if isinstance(variable_names, list) and len(variable_names) != X.shape[1]:
+        raise ValueError(
+            "The length of `variable_names` must be equal to the number of features in `X`."
+        )
+    if isinstance(X_units, list) and len(X_units) != X.shape[1]:
+        raise ValueError(
+            "The length of `X_units` must be equal to the number of features in `X`."
+        )
+    if y is not None:
+        raise ValueError("Recursive symbolic regression does not use `y`")
+    if y_units is not None:
+        raise ValueError("Recursive symbolic regression does not use `y_units`")
+    if Xresampled is not None:
+        raise ValueError("Recursive symbolic regression does not use `Xresampled`")
+
+
+class PySRSequenceRegressor(PySRRegressor):
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.recursive_history_length = recursive_history_length
+
+    def _variable_names(self, y, variable_names=None):
+        if not variable_names:
+            if y.shape[1] == 1:
+                return [f"xt_{i}" for i in range(self.recursive_history_length, 0, -1)]
+            else:
+                return [
+                    f"x{i}t_{j}"
+                    for j in range(self.recursive_history_length, 0, -1)
+                    for i in range(y.shape[1])
+                ]
+        else:
+            return [
+                i + "t_" + str(j)
+                for j in range(self.recursive_history_length, 0, -1)
+                for i in variable_names
+            ]
+
+    def fit(
+        self,
+        X,
+        y=None,
+        Xresampled=None,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+        y_units=None,
+    ) -> "PySRSequenceRegressor":
```

Comment:
```suggestion
    def fit(
        self,
        X,
        *,
        weights=None,
        variable_names: Optional[ArrayLike[str]] = None,
        complexity_of_variables: Optional[
            Union[int, float, List[Union[int, float]]]
        ] = None,
        X_units: Optional[ArrayLike[str]] = None,
    ) -> "PySRSequenceRegressor":
```

(Will also require updating the other code)

Reply:
mypy still says `Signature of "fit" incompatible with supertype "PySRRegressor"`

Reply:
Ah, I see... Hm. Maybe mypy is pointing to a design issue since the classes are fundamentally different. In other words, maybe the PySRRegressor shouldn't actually be a supertype, but rather should be an **attribute** of PySRSequenceRegressor?

Something like

```julia
class PySRSequenceRegressor(BaseEstimator):
    def __init__(recursive_history_length, **kwargs):
        self._regressor = PySRRegressor(**kwargs)
```


and then call `self._regressor...` instead of `super()...`?